homeassistant/components/pooldose/sensor.py

```diff
@@ -0,0 +1,212 @@
+        suggested_display_precision=2,
+        entity_registry_enabled_default=False,
+    ),
+    PooldoseSensorEntityDescription(
+        key="orp_calibration_type",
+        translation_key="orp_calibration_type",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        entity_registry_enabled_default=False,
+    ),
+    PooldoseSensorEntityDescription(
+        key="orp_calibration_offset",
+        translation_key="orp_calibration_offset",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        device_class=SensorDeviceClass.VOLTAGE,
+        suggested_display_precision=2,
+        entity_registry_enabled_default=False,
+    ),
+    PooldoseSensorEntityDescription(
+        key="orp_calibration_slope",
+        translation_key="orp_calibration_slope",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        device_class=SensorDeviceClass.VOLTAGE,
+        suggested_display_precision=2,
+        entity_registry_enabled_default=False,
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: PooldoseConfigEntry,
+    async_add_entities: AddConfigEntryEntitiesCallback,
+) -> None:
+    """Set up Pooldose sensor entities from a config entry."""
+    coordinator = config_entry.runtime_data.coordinator
+    client = config_entry.runtime_data.client
+    serial_number = config_entry.unique_id
+    if not serial_number:
+        _LOGGER.error("No serial number found in config entry, cannot create entities")
+        return
+    device_properties = config_entry.runtime_data.device_properties
+
+    available = client.available_sensors()
+    entities: list[SensorEntity] = []
+
+    for description in SENSOR_DESCRIPTIONS:
+        if description.key not in available:
+            continue
+        entities.append(
+            PooldoseSensor(
+                coordinator,
+                description,
+                serial_number,
+                device_properties,
+            )
+        )
+
+    async_add_entities(entities)
+
+
+class PooldoseSensor(PooldoseEntity, SensorEntity):
+    """Sensor entity for the Seko Pooldose API."""
+
+    entity_description: PooldoseSensorEntityDescription
+
+    def __init__(
+        self,
+        coordinator: PooldoseCoordinator,
+        description: PooldoseSensorEntityDescription,
+        serial_number: str,
+        device_properties: dict[str, Any],
+    ) -> None:
+        """Initialize a Pooldose sensor entity."""
+        self.entity_description = description
+        super().__init__(
+            coordinator,
+            serial_number,
+            device_info(device_properties, serial_number),
+        )
+
+    @property
+    def native_value(self) -> float | int | str | None:
+        """Return the current value of the sensor."""
+        if not self.coordinator.data:
+            return None
+
+        data = self.coordinator.data
+        sensor_data = data.get(self.entity_description.key)
+        if not sensor_data:
+            return None
```

Comment:
I assume that if we can't get sensor data, there is something wrong with the retrieval. Would it make sense to add an `available` property that checks `super().available` and that the entity description key is present in the coordinator data?